src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -444,11 +446,29 @@ protected function getEnv($name)
         if (false !== $env = getenv($name)) {
             return $this->envCache[$name] = $env;
         }
-        if (!$this->hasParameter("env($name)")) {
+        if (!$this->isCompiled() && $this->parameterBag instanceof EnvPlaceholderParameterBag) {
```

Comment:
Im not sure; isnt getEnv() always about getting real env values? If it's a EnvPlaceholderParameterBag we always need to get real value from parent::get.

Doing this from ContainerBuilder::getEnv will be a pain.